### PR TITLE
TS-4263: Global key block configurable via Records.config

### PR DESCRIPTION
**Pull request by persiaAziz**

### CODE REVIEW

**Comment by jpeach** [comment]
[approve ci]

**Comment by atsci** [comment]
Linux build *failed*! See https://ci.trafficserver.apache.org/job/Github-Linux/535/ for details.

**Comment by zwoop** [comment]
make clang-format

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/639/ for details.

**Comment by persiaAziz** [comment]
Thanks @zwoop . I was using a tool for clang-formatting. 
Please approve

**Comment by zwoop** [comment]
[approve ci]

File: iocore/net/SSLUtils.cc [hunk 1]
```diff
@@ -95,6 +95,8 @@ struct ssl_user_config {
   ssl_user_config() : opt(SSLCertContext::OPT_NONE)
   {
     REC_ReadConfigInt32(session_ticket_enabled, "proxy.config.ssl.server.session_ticket.enable");
+    REC_ReadConfigStringAlloc(ticket_key_filename, "proxy.config.ssl.server.ticket_key.filename");
+    Debug("ssl", "ticket  key filename %s", (const char *)ticket_key_filename);
   }
   int session_ticket_enabled; // ssl_ticket_enabled - session ticket enabled
```
Comment (zwoop) [inline iocore/net/SSLUtils.cc]: I know it's not in your patch, but can you fix the line where we declare ticket_key_filename such that clang-format doesn't wrap the line? I.e. make the comment shorter (or multiple lines). It just looked really odd, so good time to clean that up here :).

File: iocore/net/SSLUtils.cc [hunk 2]
```diff
@@ -544,7 +547,64 @@ ssl_context_enable_ecdh(SSL_CTX *ctx)
 
   return ctx;
 }
+static ssl_ticket_key_block *
+ssl_create_ticket_keyblock(const char *ticket_key_path)
```
Comment (zwoop) [inline iocore/net/SSLUtils.cc]: Hmmm, maybe I'm missing something, but it seems most of this code is duplicated from ssl_context_enable_tickets() ? Can we not refactor this such that we don't get such massive code duplication?

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/641/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/537/ for details.

File: iocore/net/SSLUtils.cc (hunk 2 above)
Comment (persiaAziz) [inline iocore/net/SSLUtils.cc]: Yes most of the code are duplicated from the ssl_context_enable_tickets, but in this function I am just creating the key block  not enabling session ticket for any particular context, So I decided to make a separate function. What we can do is have ssl_context_enable_ticket call ssl_create_ticket_keyblock. That way the code will look cleaner

File: iocore/net/SSLUtils.cc (hunk 1 above)
Comment (persiaAziz) [inline iocore/net/SSLUtils.cc]: Sure :)

File: iocore/net/SSLUtils.cc (hunk 2 above)
Comment (zwoop) [inline iocore/net/SSLUtils.cc]: Exactly. :)

**Comment by persiaAziz** [comment]
Closing this PR since I messed up my branch with git [AGAIN!]

**Pull request closed without merging**